deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,239 @@
+
+        if alternative_metrics is not None and len(alternative_metrics) == 0:
+            raise ValueError('alternative_metrics - expected to receive not empty dict of scorers!')
+
+        elif alternative_metrics is not None:
+            self.alternative_metrics = {}
+
+            for name, metric in alternative_metrics.items():
+                if isinstance(metric, t.Callable):
+                    self.alternative_metrics[name] = metric
+                elif isinstance(metric, str):
+                    self.alternative_metrics[name] = get_scorer(metric)
+                else:
+                    raise ValueError(
+                        f"alternative_metrics - expected to receive 'Mapping[str, Callable]' but got " #pylint: disable=inconsistent-quotes
+                        f"'Mapping[str, {type(metric).__name__}]'!" #pylint: disable=inconsistent-quotes
+                    )
+
+    def run(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        """Run Check.
+
+        Args:
+            dataset (Dataset): a dataset object
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance
+
+        Returns:
+            CheckResult: value is a dictionary of labels, value of each key is another dictionary
+                with calculated metrics
+
+        Raises:
+            DeepchecksValueError:
+                if task type is not binary or multi-class;
+                if dataset does not have a label column or the label dtype is unsupported;
+                if provided dataset is empty;
+        """
+        return self._class_performance_imbalance(dataset, model)
+
+    def _class_performance_imbalance(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        expected_model_types = [ModelType.BINARY, ModelType.MULTICLASS]
+
+        dataset.validate_label(check_name)
+        dataset.validate_features(check_name)
+        Dataset.validate_dataset(dataset, check_name)
+        task_type_validation(model, dataset, expected_model_types, check_name)
+
+        labels = t.cast(pd.Series, dataset.label_col())
+        features = t.cast(pd.DataFrame, dataset.features_columns())
+
+        if self.alternative_metrics is not None:
+            df = pd.DataFrame.from_dict(self._execute_alternative_metrics(
+                model, features, labels
+            ))
+        else:
+            y_true = labels
+            y_pred = model.predict(features)
+            unique_labels = get_unique_labels(labels, y_pred)
+            df = pd.DataFrame.from_dict({
+                name: dict(zip(unique_labels, metric_func(y_true, y_pred, labels=unique_labels))) # type: ignore
+                for name, metric_func in self._get_default_metrics().items()
+            })
```

Comment:
why not initalize dict like alternative_metrics and then have single 
```
df = pd.DataFrame.from_dict(self._execute_alternative_metrics(
                model, features, labels
            ))
```

Reply:
we want the value to be in the format `{'label_name': {'metric_name': <value>}}`, without `DataFrame` we would need to make some extra work to achieve this.

What I mean is that with help of `DataFrame.transpose` implementation looks cleaner and more readable.

Reply:
I didn't understand your comment. I'm using dataframe in the example I wrote. I'm saying to reuse the `_execute_alternative_metrics` also in the else case.

Reply:
oh, sorry I did not get you.

User-provided and our default metrics are different.
UserProvided metrics must have signature `Callable[[model, features, labels], Dict[str, int|float]]`
whereas our default metrics have signature `Callable[[y_true, y_pred], np.ndarray]`

therefore we process thems differently

Reply:
hm/ it will not work

sorry, I answered one of the Matan questions previously incorrectly,
there is no scorer that returns dict if you pass `average=None`
they return only numbers or np.array

`classification_report` can return dict but not scorer functions

so, as a result:

1. 
there is actually no sense in passing the names of the sklearn scorers to the check, because we will not be able to use them`s to calculate metrics per class

2. 
in order to build a dict of values per class we need to zip list of unique labels with an array returned by the scorer, in the same way how it is done in the 'classification_report' function

from 'classification_report' (lines 2107-2116):

```
labels = ... # list of labels
target_names = ... # stringified list of labels
p, r, f1, s = precision_recall_fscore_support(
    y_true,
    y_pred,
    labels=labels,
    average=None,
    sample_weight=sample_weight,
    zero_division=zero_division,
)
rows = zip(target_names, p, r, f1, s)
```

Reply:
so instead of using scorer that returns dict, maybe it better to use regular scorer and for each class transform it to 0 and 1:
```
for label in labels:
  df['target'].map(lambda x: 1 if x == label else 0)
  # calc score for label...
```

Reply:
so, did I get you right that to calculate a metric for some specific class it is enough to have data (labels/predictions) only with that class? I mean that I do not need an entire dataset to calc metric for some specific class? and that it is true for any existing metric?

Reply:
I think so, but I don't see a reason to filter out rows with other target/predictions, it doesn't make it easier.
@noamzbr is that right?

Reply:
Look at this [thread]( https://stats.stackexchange.com/questions/51296/how-do-you-calculate-precision-and-recall-for-multiclass-classification-using-co) to better understand.
Short version - it's calculated per class using the appropriate row in the confusion matrix, which is in turn calculated on *all* samples. You can see an example in [precision_recall_fscore_support](https://github.com/scikit-learn/scikit-learn/blob/0d378913b/sklearn/metrics/_classification.py#L1370). It's true that samples that were both not predicted as the inspected class AND their label is not the inspected class don't contribute to the calculation of per class precision/recall.

I'm for the approach @yromanyshyn suggested - zipping the labels with the np.array

Reply:
the zip sounds like technical stuff, the problem is implementing a function like  `precision_recall_fscore_support` which use scorer to calculate each class separatly

Reply:
Another option is to use `utils.metrics.get_metrics_list` and add parameter to it "average". than of course we will have to change the rest of the implementation to use `make_scorer` with this param

Reply:
>he zip sounds like technical stuff, the problem is implementing a function like precision_recall_fscore_support which use scorer to calculate each class separatly

I did not get you, You mean that it will be uneasy for the users to impl custom scorers for the check?

> Another option is to use utils.metrics.get_metrics_list and add parameter to it "average". than of course we will have to change the rest of the implementation to use make_scorer with this param

1. after we wrap the scorer function into `make_scorer` we cannot call it anymore with the `average` parameter, so we will not be able to use scorers returned by the `utils.metrics.get_metrics_list` whereas they are initialized with default `average` or with `average='macro'`
2. I do not think that we need `average=None` everywhere/

Reply:
1. that's why I said "we will have to change the rest of the implementation to use make_scorer with this param". we will create it on the fly

Reply:
I updated implementation